File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n=nums.size();
        int maxSum=INT_MIN;
        int current_sum=0;
        for (int i=0;i<n;i++)
    {
        current_sum +=nums[i];
        if (current_sum>maxSum)
        {
            maxSum=current_sum;
        }
        if(current_sum<0)
    {
        current_sum=0;
    }
    }

return maxSum;        
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {

        int count =0;
        int maxcount=Integer.MIN_VALUE;
        for (int i=0;i<nums.length;i++)
        {
            count= count+nums[i];
            if (count>maxcount)
            {
                maxcount=count;
            }
            if (count<0)
            {
                count =0;
            }
            
        }
        return maxcount;
        
    }
}